app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -1,20 +1,98 @@
 package com.jetbrains.handson.mpp.mobile
 
-import androidx.appcompat.app.AppCompatActivity
+import android.app.AlertDialog
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
-import android.widget.TextView
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemSelectedListener
+import android.widget.Button
+import android.widget.Spinner
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.departSelector
+
 
 class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
+    private var departStation: Station? = null
+    private var arrivalStation: Station? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
         val presenter = ApplicationPresenter()
         presenter.onViewTaken(this)
+
+        val stationsHelper = StationsHelper()
+
+        val stations = stationsHelper.stations
+
+        populateSpinner(R.id.departSelector, stations)
+        populateSpinner(R.id.arrivalSelector, stations)
+
+        findViewById<Spinner>(R.id.departSelector).setOnItemSelectedListener(object : OnItemSelectedListener {
+            override fun onItemSelected(
+                parentView: AdapterView<*>?,
+                selectedItemView: View?,
+                position: Int,
+                id: Long
+            ) {
+                departStation = stations[position]
+                println(departStation)
+            }
+
+            override fun onNothingSelected(parentView: AdapterView<*>?) {
+                // your code here
+            }
+        })
+
+        findViewById<Spinner>(R.id.arrivalSelector).setOnItemSelectedListener(object : OnItemSelectedListener {
+            override fun onItemSelected(
+                parentView: AdapterView<*>?,
+                selectedItemView: View?,
+                position: Int,
+                id: Long
+            ) {
+                arrivalStation = stations[position]
+            }
+
+            override fun onNothingSelected(parentView: AdapterView<*>?) {
+                // your code here
+            }
+        })
+
+        findViewById<Button>(R.id.submitButton)
+            .setOnClickListener {
+                if(departStation == null || arrivalStation == null) {
+                    val builder = AlertDialog.Builder(this)
+                    builder.setTitle("Androidly Alert")
+                    builder.setMessage("We have a message")
+                    builder.setPositiveButton(android.R.string.yes) { dialog, which ->
+                        Toast.makeText(applicationContext,
+                            android.R.string.yes, Toast.LENGTH_SHORT).show()
+                    }
```

Comment:
Rather than instantiating and then calling the builder, you can use an `.apply` method to do it all at once, and it looks nicer imo :) 

```
        AlertDialog.Builder(this).apply {
            setTitle("Androidly Alert")
            setMessage("We have a message")
            setPositiveButton(android.R.string.yes) { dialog, which ->
                Toast.makeText(applicationContext, android.R.string.yes, Toast.LENGTH_SHORT).show()
            }
```